common/internal_model/src/test/scala/uk/ac/wellcome/models/index/IndexConfigFieldsTest.scala

```diff
@@ -0,0 +1,105 @@
+package uk.ac.wellcome.models.index
+
+import org.scalatest.concurrent.Eventually
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import uk.ac.wellcome.json.utils.JsonAssertions
+import uk.ac.wellcome.models.work.generators.WorkGenerators
+import weco.catalogue.internal_model.generators.ImageGenerators
+import uk.ac.wellcome.elasticsearch.IndexConfig
+import com.sksamuel.elastic4s.ElasticDsl._
+import com.sksamuel.elastic4s.requests.mappings.dynamictemplate.DynamicMapping
+import io.circe._, io.circe.generic.semiauto._, io.circe.syntax._
+import com.sksamuel.elastic4s._
+import com.sksamuel.elastic4s.requests.searches._
+
+case class TestDoc(label: String)
+object TestDoc {
+  implicit val fooDecoder: Decoder[TestDoc] = deriveDecoder
+  implicit val fooEncoder: Encoder[TestDoc] = deriveEncoder
```

Comment:
```suggestion
  implicit val testDocDecoder: Decoder[TestDoc] = deriveDecoder
  implicit val testDocEncoder: Encoder[TestDoc] = deriveEncoder
```
If you import JsonUtils I think these will be fully automatically derived

Reply:
Over time auto-derivation slows things down like crazy so trying not to use it.